tests/containers/podman_quadlet.pm

```diff
@@ -67,30 +102,52 @@ _EOF_
     # start the generator and check whether the files are generated
     systemctl("daemon-reload");
     check_unit_states();
-    systemctl("is-active sleeper.service", expect_false => 1);
-    systemctl("is-active sleeper-volume.service", expect_false => 1);
+    for my $unit (@units) {
+        systemctl("is-active $unit", expect_false => 1);
+    }
+
+    # start build unit
+    systemctl("start $unit_name-build.service");
+    record_info('Build output', script_output("journalctl --no-pager -u $unit_name-build"));
+
+    # validate image build
+    systemctl("is-active $unit_name-build.service");
+    validate_script_output('podman images -n', qr/$build_imagetag/);
 
     # start the container
-    systemctl("start sleeper-volume.service");
-    systemctl("start sleeper.service");
+    systemctl("start $unit_name.service");
     check_unit_states();
-    systemctl("is-active sleeper.service");
-    systemctl("is-active sleeper-volume.service");
+    systemctl("is-active $unit_name.service");
+    systemctl("is-active $unit_name-pod.service");
+    systemctl("is-active $unit_name-volume.service");
+    systemctl("is-active $unit_name-network.service");
+
+    # check if the network exists on host
+    my $podman_net = JSON->new->decode(script_output("podman network inspect systemd-$unit_name"));
+    my $net_if = $podman_net->[0]->{network_interface};
```

Comment:
You can get this information directly from podman without parsing JSON in Perl like this:

```suggestion
    my $net_if = script_output("podman network inspect -f '{{ .NetworkInterface }}' systemd-$unit_name");
```